libs/execution/tests/unit/executor_parameters_dispatching.cpp

```diff
@@ -0,0 +1,477 @@
+    }
+
+    template <typename Executor>
+    void mark_end_of_scheduling(Executor&&)
+    {
+        ++params_count;
+    }
+
+    template <typename Executor>
+    void mark_end_execution(Executor&&)
+    {
+        ++params_count;
+    }
+};
+
+namespace hpx { namespace parallel { namespace execution {
+    template <>
+    struct is_executor_parameters<test_begin_end> : std::true_type
+    {
+    };
+}}}    // namespace hpx::parallel::execution
+
+///////////////////////////////////////////////////////////////////////////////
+void test_mark_begin_execution()
+{
+    {
+        params_count = 0;
+        exec_count = 0;
+
+        hpx::parallel::execution::mark_begin_execution(
+            test_begin_end{}, hpx::parallel::execution::par.executor());
+
+        HPX_TEST_EQ(params_count, std::size_t(1));
+        HPX_TEST_EQ(exec_count, std::size_t(0));
+    }
+
+    {
+        params_count = 0;
+        exec_count = 0;
+
+        hpx::parallel::execution::mark_begin_execution(
+            test_begin_end{}, test_executor_begin_end{});
+
+        HPX_TEST_EQ(params_count, std::size_t(0));
+        HPX_TEST_EQ(exec_count, std::size_t(1));
+    }
+}
+
+void test_mark_end_of_scheduling()
+{
+    {
+        params_count = 0;
+        exec_count = 0;
+
+        hpx::parallel::execution::mark_end_of_scheduling(
+            test_begin_end{}, hpx::parallel::execution::par.executor());
+
+        HPX_TEST_EQ(params_count, std::size_t(1));
+        HPX_TEST_EQ(exec_count, std::size_t(0));
+    }
+
+    {
+        params_count = 0;
+        exec_count = 0;
+
+        hpx::parallel::execution::mark_end_of_scheduling(
+            test_begin_end{}, test_executor_begin_end{});
+
+        HPX_TEST_EQ(params_count, std::size_t(0));
+        HPX_TEST_EQ(exec_count, std::size_t(1));
+    }
+}
+
+void test_mark_end_execution()
+{
+    {
+        params_count = 0;
+        exec_count = 0;
+
+        hpx::parallel::execution::mark_end_execution(
+            test_begin_end{}, hpx::parallel::execution::par.executor());
+
+        HPX_TEST_EQ(params_count, std::size_t(1));
+        HPX_TEST_EQ(exec_count, std::size_t(0));
+    }
+
+    {
+        params_count = 0;
+        exec_count = 0;
+
+        hpx::parallel::execution::mark_end_execution(
+            test_begin_end{}, test_executor_begin_end{});
+
+        HPX_TEST_EQ(params_count, std::size_t(0));
+        HPX_TEST_EQ(exec_count, std::size_t(1));
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+int hpx_main()
+{
+    test_get_chunk_size();
+    test_maximal_number_of_chunks();
+    test_reset_thread_distribution();
+    test_processing_units_count();
+    test_mark_begin_execution();
+    test_mark_end_of_scheduling();
+    test_mark_end_execution();
+
+    return hpx::finalize();
+}
+
+int main(int argc, char* argv[])
+{
+    // By default this test should run on all available cores
+    std::vector<std::string> const cfg = {"hpx.os_threads=all"};
```

Comment:
Does this really need `all`? That means hyperthreads as well. `cores` (default) or `4` would possibly be enough?

Reply:
We've been using this in many tests for a long time. This is just a copy&paste of some related test, I believe. I think nowadays using all cores is the default anyways, so we can go ahead and remove such configurations from all places.

Reply:
Then let's keep it like this. We can go and change it everywhere afterwards if we feel like it.